[Windows 10 N/KN] Matrix RC startup raises mfplat.dll missing -> Media Feature Pack not installed

## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Yesterday I've upgraded by HTPC to Matrix RC after some testing on my laptop where everything worked fine. Installation went fine but when starting Kodi the error 'mfplat.dll missing' and the program wouldn't start. 


## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Regular start of Kodi Matrix after upgrading from v18.


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Error message mfplat.dll missing, program won't start.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
It seems that my HTPC is running a Windows 10 N version, which don't come with the Media Feature Pack. Manually installing this (pre 1903 versions) or enabling it via 'turn windows features on or off' and rebooting the computer fixed the issue.

The Media Feature Pack provides the missing dll file but it not installed on Windows 10 N or KN versions.



### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. Install Matrix on a Windows 10 N or KN version without Media Feature Pack already enabled
2. Start Kodi



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [x] Windows **N version**
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: Matrix RC



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

> only `mediafoundation` should be disabled, `amf` isn't used actually. see the log from latest build https://jenkins.kodi.tv/job/WIN-64/15943/consoleFull
> 
> ```
> 16:36:46 External libraries:
> 16:36:47 libdav1d                openssl
> 16:36:47 mediafoundation         zlib
> ```

I just grepped FFmpeg for the missing symbols from the FFmpeg bump PR.

```
14:25:16      Creating library F:/builds/workspace/WIN-64/kodi-build.x64/Release/kodi.lib and object F:/builds/workspace/WIN-64/kodi-build.x64/Release/kodi.exp
14:25:16 avcodec.lib(mfenc.o) : error LNK2019: unresolved external symbol MFCreateMediaType referenced in function mf_choose_output_type [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mfenc.o) : error LNK2019: unresolved external symbol IID_ICodecAPI referenced in function mf_init [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mfenc.o) : error LNK2019: unresolved external symbol IID_IMFMediaEventGenerator referenced in function mf_unlock_async [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mf_utils.o) : error LNK2019: unresolved external symbol MFStartup referenced in function ff_instantiate_mf [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mf_utils.o) : error LNK2019: unresolved external symbol MFShutdown referenced in function ff_free_mf [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mf_utils.o) : error LNK2019: unresolved external symbol MFCreateAlignedMemoryBuffer referenced in function ff_create_memory_sample [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mf_utils.o) : error LNK2019: unresolved external symbol MFCreateSample referenced in function ff_create_memory_sample [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 avcodec.lib(mf_utils.o) : error LNK2019: unresolved external symbol IID_IMFTransform referenced in function ff_instantiate_mf [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
14:25:16 F:\builds\workspace\WIN-64\kodi-build.x64\Release\kodi.exe : fatal error LNK1120: 8 unresolved externals [F:\builds\workspace\WIN-64\kodi-build.x64\kodi.vcxproj]
```

so
```
MFCreateMediaType 
IID_ICodecAPI 
IID_IMFMediaEventGenerator
MFStartup 
MFShutdown 
MFCreateAlignedMemoryBuffer 
MFCreateSample 
IID_IMFTransform 
```

results in
```
$ git grep -E "(MFCreateMediaType|IID_ICodecAPI|IID_IMFMediaEventGenerator|MFStartup|MFShutdown|MFCreateAlignedMemoryBuffer|MFCreateSample|IID_IMFTransform)"
configure:mediafoundation_deps="mftransform_h MFCreateAlignedMemoryBuffer"
configure:check_func_headers mfapi.h MFCreateAlignedMemoryBuffer -lmfplat
libavcodec/mf_utils.c:    hr = MFCreateSample(&sample);
libavcodec/mf_utils.c:    hr = MFCreateAlignedMemoryBuffer(size, align - 1, &buffer);
libavcodec/mf_utils.c:    hr = MFStartup(MF_VERSION, MFSTARTUP_FULL);
libavcodec/mf_utils.c:    MFShutdown();
libavcodec/mf_utils.c:        hr = IMFActivate_ActivateObject(activate[n], &IID_IMFTransform,
libavcodec/mfenc.c:        hr = MFCreateMediaType(&out_type);
libavcodec/mfenc.c:    hr = IMFTransform_QueryInterface(c->mft, &IID_IMFMediaEventGenerator, (void **)&c->async_events);
libavcodec/mfenc.c:    hr = IMFTransform_QueryInterface(c->mft, &IID_ICodecAPI, (void **)&c->codec_api);
```

```
$ git grep " mf_utils.o"
libavcodec/Makefile:OBJS-$(CONFIG_AAC_MF_ENCODER)          += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_AC3_MF_ENCODER)          += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_H264_MF_ENCODER)         += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_HEVC_MF_ENCODER)         += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_MP3_MF_ENCODER)          += mfenc.o mf_utils.o
$ git grep " mfenc.o"
libavcodec/Makefile:OBJS-$(CONFIG_AAC_MF_ENCODER)          += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_AC3_MF_ENCODER)          += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_H264_MF_ENCODER)         += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_HEVC_MF_ENCODER)         += mfenc.o mf_utils.o
libavcodec/Makefile:OBJS-$(CONFIG_MP3_MF_ENCODER)          += mfenc.o mf_utils.o
```

So yes, Look like just mediafoundation. Not sure where I saw AMF.

--- a/cmake/scripts/windows/ArchSetup.cmake
+++ b/cmake/scripts/windows/ArchSetup.cmake
@@ -73,7 +73,7 @@ link_directories(${DEPENDENCIES_DIR}/lib)
 # Additional libraries
 list(APPEND DEPLIBS bcrypt.lib d3d11.lib DInput8.lib DSound.lib winmm.lib Mpr.lib Iphlpapi.lib WS2_32.lib
                     PowrProf.lib setupapi.lib Shlwapi.lib dwmapi.lib dxguid.lib DelayImp.lib version.lib
-                    crypt32.lib Mfplat.lib Mfuuid.lib Strmiids.lib)
+                    crypt32.lib)
 
 # NODEFAULTLIB option
 set(_nodefaultlibs_RELEASE libcmt)

--- a/cmake/scripts/windowsstore/ArchSetup.cmake
+++ b/cmake/scripts/windowsstore/ArchSetup.cmake
@@ -91,8 +91,7 @@ set(gtest_force_shared_crt ON CACHE STRING "" FORCE)
 link_directories(${MINGW_LIBS_DIR}/lib
                  ${DEPENDENCIES_DIR}/lib)
 
-list(APPEND DEPLIBS bcrypt.lib d3d11.lib WS2_32.lib dxguid.lib dloadhelper.lib WindowsApp.lib
-                    Mfplat.lib Mfuuid.lib Strmiids.lib)
+list(APPEND DEPLIBS bcrypt.lib d3d11.lib WS2_32.lib dxguid.lib dloadhelper.lib WindowsApp.lib)
 
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WINMD:NO")
 set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} /NODEFAULTLIB:msvcrt /DEBUG:FASTLINK /OPT:NOREF /OPT:NOICF")

--- a/tools/buildsteps/windows/buildffmpeg.sh
+++ b/tools/buildsteps/windows/buildffmpeg.sh
@@ -84,6 +84,7 @@ do_addOption "--disable-gnutls"
 do_addOption "--enable-openssl"
 do_addOption "--enable-nonfree"
 do_addOption "--toolchain=msvc"
+do_addOption "--disable-mediafoundation"
 
 if [ "$ARCH" == "x86_64" ]; then
   FFMPEG_TARGET_OS=win64

